include/numerics/tensor_tools.h

```diff
@@ -99,6 +99,10 @@ inline
 auto norm_sq(const VectorValue<T> & a) -> decltype(std::norm(T()))
 {return a.norm_sq();}
 
+template<typename T>
+inline
+auto is_zero(const T & a){ return a.is_zero();}
```

Comment:
Return type should just be `bool`. IIRC automatic return type deduction is a C++14 thing. The other places where it is used in this class are for trailing return types.

Reply:
You can do automatic return type deduction in C++11 but only with that awful redundant trailing arrow notation, not with just a single "auto".

Reply:
Good, I don't like all the auto stuff anyway.